thorlcr/activities/indexread/thindexreadslave.cpp

```diff
@@ -763,11 +768,9 @@ class CIndexReadSlaveActivity : public CIndexReadSlaveBase
                 if (needTransform)
                 {
                     size32_t sz = helper->transform(ret, r);
+                    callback.finishedRow();
```

Comment:
logically I would like to put this call in an exit scope (COnScopeExit) just before the canMatch, as anything after it can use it, and would make the code safer and cleaner.
But I think it (and it's use of std::function's) does add a fair amount of overhead, including heap allocation from what I've read), so I avoid it.